### PR TITLE
Removes Clone Damage

### PR DESCRIPTION
**Pull request by strandsofivy**
## About The Pull Request

Does what it says on the tin. We don't have any "special" sources of clone damage left in the game, most of them are rather trivial so I bunched them together into this PR.

Notable things removed:
- Clonexadone, because its entire thing was centered around clone damage
- Decloner gun, it's also centered around cloning damage, I couldn't think of a replacement mechanic and nobody uses it anyways
- Everything else already dealt clone damage as a side (rainbow knife deals a random damage type for example), so these sources were removed

## Why It's Good For The Game

Consider the four sources of normal damage that you can get: Brute, Burn, Toxins and Oxygen. These four horsemen of the apocalypse are very well put together and it's no surprise that they are in the game, as you can fit any way of damaging a mob into them. Getting beaten to death by a security officer? Brute damage. Running around on fire? Burn damage. Poisoned or irradiated? Toxin damage. Suffocating in space? Brute, burn and oxygen damage. Technically there's also stamina damage but that's its own ballpark and it also makes sense why we have a damage number for it. 

Picture this now: We have this cool mechanic called "clone pods" where you can magically revive dead people with absolute ease. We don't want it to be for free though, it comes at a cost. This cost is clone damage, and it serves to restrain people from abusing cloning. 

Fast forward time a bit and cloning is now removed from the game. What stays with us is a damage number that is intrinsically tied to the context of a removed feature. It was a good idea that we had it for that feature at the time, but now it just sits there. It's the odd one out from all the other damage types. You can easily explain why your blade dealt brute damage, but how are you going to fit clone damage into any context without also becoming extremely specific? 

My point is: **clone damage is conceptually a flawed mechanic because it is too specific**. That is the major issue why no one uses it, and why that makes it unworthy of being a damage stat. 
Don't take my word for it though, because a while ago we only had a handful of sources for this damage type in the game. And in most of the rounds where you saw this damage, it came from only one department. It's not worthwhile to keep it around as a damage number. People also didn't know what to do with this damage type, so we currently have two ways of healing clone damage: Cryotubes as a roundstart way of healing clone damage and Rezadone, which instantly sets your clone damage to 0 on the first tick. As a medical doctor, when was the last time you saw someone come in with clone damage and thought to yourself, "Oh, this person has clone damage, I cannot wait to heal them!" ?

Now we have replacements for these clone damage sources. Slimes? Slime status effect that deals brute instead of clone. Cosmic heretics? Random organ damage, because their mechanics are already pretty fleshed out. Decloning virus? The virus operated as a "ticking timebomb" which used cloning damage as the timer, so it has been reworked to not use clone damage. What remains after all this is now a basically unused damage type. Every specific situation that used clone damage is now relying on another damage type. Now it's time to put clone damage to rest once and for all.

Sure, you can technically add some form of cellular degradation in the future, but it shouldn't be a damage number. The idea of your cells being degraded is a cool concept, don't get me wrong, but make it a status effect or maybe even a wound for that matter. 

## Changelog

:cl:
del: Removed clone damage.
del: Removed the decloner gun.
del: Removed clonexadone.
/🆑

### CODE REVIEW

**Comment by EricZilla** [comment]
How is the paradox clone supposed to kill the orginal now - oh wait

GOOD JOB KILL IT

**Comment by SapphoQueer** [comment]
most based pr of 2023

**Comment by Vect0r2** [comment]
I really liked the decloner :(

**Comment by Higgin** [comment]
Design-wise, we've kinda been slowly walking up to the line with this one for a bit. This is naturally where it ends.

I've got mixed feelings. Having some things that you **have** to have other departments or other specific systems for can be used for an effect (i.e. - you NEED med and cryo/clonex to really deal with clone damage in most cases,(, but it also makes those things into massively worse pain-points when you cannot access their niche solution (one of the big problems with viro that I've tried to address.)

This was long overdue, but I hope we don't totally lean out from "stickier" systems even if clone damage was one of the least intuitive and least engaging ones there was.

It'd be cool to have a gun that just irradiates on hit (no, not the ebow, the actual rad status effect.)

File: code/__DEFINES/combat.dm [hunk 1]
```diff
@@ -56,7 +54,6 @@
 #define BRUTELOSS (1<<0)
 #define FIRELOSS (1<<1)
 #define TOXLOSS (1<<2)
-#define CLONELOSS (1<<3)
 #define OXYLOSS (1<<4)
 #define STAMINALOSS (1<<5)
 #define SHAME (1<<6)
```
Comment (san7890) [inline code/__DEFINES/combat.dm:59]: you need to shift all the bitflags past where cloneloss was down by one, cant' have it empty

**Review by san7890 — CHANGES_REQUESTED** [review]

**Comment by SuperNovaa41** [comment]
We discussed this in coding-gen a bit, its an exotic damage type that gives people a reason to go to the medbay, instead of being able to fix it on their own. It comes from very few sources so it's a niche way of lowering the max health cap without an easy immediate fix. I think that if we are to keep it though, we should add more sources of clone damage to make it more of a "hey this isnt used lets remove it" feature anymore though.
If another maint disagrees feel free to remove DNM

**Comment by MrMelbert** [comment]
> If another maint disagrees feel free to remove DNM

I know you haven't been around for a while but cloning damage has been actively removed from a lot of things, at this point it's effectively a dead damage type.

**Comment by SuperNovaa41** [comment]
> Wounds have been pretty underdeveloped anyways, we can totally add more like "internal bleeding" or wounds that raises damage taken or adds a minimum level of brute/burn (lowering effective health cap) of a limb as methods of difficult-to-fix / medbay-help-required damage

Thats definitely a far better direction

**Comment by optimumtact** [comment]
```

MariaDB [tgstation13]> describe death
    -> ;
+-------------+----------------------+------+-----+---------+----------------+
| Field       | Type                 | Null | Key | Default | Extra          |
+-------------+----------------------+------+-----+---------+----------------+
| id          | int(11)              | NO   | PRI | NULL    | auto_increment |
| pod         | varchar(50)          | NO   |     | NULL    |                |
| coord       | varchar(32)          | YES  |     | NULL    |                |
| x_coord     | smallint(5) unsigned | NO   |     | NULL    |                |
| y_coord     | smallint(5) unsigned | NO   |     | NULL    |                |
| z_coord     | smallint(5) unsigned | NO   |     | NULL    |                |
| mapname     | varchar(32)          | NO   |     | NULL    |                |
| server_ip   | int(10) unsigned     | NO   |     | NULL    |                |
| server_port | smallint(5) unsigned | NO   |     | NULL    |                |
| round_id    | int(11) unsigned     | YES  | MUL | NULL    |                |
| tod         | datetime             | NO   | MUL | NULL    |                |
| job         | varchar(32)          | NO   |     | NULL    |                |
| special     | varchar(32)          | YES  |     | NULL    |                |
| name        | varchar(96)          | NO   |     | NULL    |                |
| byondkey    | varchar(32)          | NO   | MUL | NULL    |                |
| laname      | varchar(96)          | YES  |     | NULL    |                |
| lakey       | varchar(32)          | YES  |     | NULL    |                |
| bruteloss   | smallint(5) unsigned | NO   |     | NULL    |                |
| brainloss   | smallint(5) unsigned | NO   |     | NULL    |                |
| fireloss    | smallint(5) unsigned | NO   |     | NULL    |                |
| oxyloss     | smallint(5) unsigned | NO   |     | NULL    |                |
| toxloss     | smallint(5) unsigned | NO   |     | NULL    |                |
| cloneloss   | smallint(5) unsigned | YES  |     | 0       |                |
| staminaloss | smallint(5) unsigned | NO   |     | NULL    |                |
| last_words  | varchar(255)         | YES  |     | NULL    |                |
| suicide     | tinyint(4)           | NO   |     | 0       |                |
+-------------+----------------------+------+-----+---------+----------------+
26 rows in set (0.817 sec)

MariaDB [tgstation13]> select * from schema_revision order by date desc limit 1;
+-------+-------+---------------------+
| major | minor | date                |
+-------+-------+---------------------+
|     5 |    26 | 2023-12-04 21:08:55 |
+-------+-------+---------------------+
1 row in set (0.154 sec)
```

**Pull request merged**